src/twisted/web/http.py

```diff
@@ -236,6 +236,58 @@
 monthname_lower = [name and name.lower() for name in monthname]
 
 
+def _parseRequestLine(line: bytes) -> tuple[bytes, bytes, bytes]:
+    """
+    Parse an HTTP request line, which looks like:
+
+        GET /foo/bar HTTP/1.1
+
+    This function attempts to validate the well-formedness of
+    the line. RFC 9112 section 3 provides this ABNF:
+
+        request-line   = method SP request-target SP HTTP-version
+
+    We allow any method that is a valid token:
+
+        method         = token
+        token          = 1*tchar
+        tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+                        / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+                        / DIGIT / ALPHA
+
+    We allow any non-empty request-target that contains only printable
+    ASCII characters (no whitespace).
+
+    The RFC defines HTTP-version like this:
+
+        HTTP-version  = HTTP-name "/" DIGIT "." DIGIT
+        HTTP-name     = %s"HTTP"
+
+    However, this function is more strict than the RFC: we only allow
+    HTTP versions of 1.0 and 1.1, as later versions of HTTP don't use
+    a request line.
+
+    @returns: C{(method, request, version)} three-tuple
+
+    @raises: L{ValueError} when malformed
+    """
+    method, request, version = line.split(b" ")
```

Comment:
Do we have a test that checks for extra spaces?

In the previous code we were using `line.split(sep=None)` which has a different behaviour than `sep=b' '`

```
>>> ca = 'some   text  here'
>>> ca.split()
['some', 'text', 'here']
>>> ca.split(' ')
['some', '', '', 'text', '', 'here']
```

----------

I think that this change is ok as the RFC asks for a single `SP` (space), but maybe it would help to have an explicit test ... in the case this is refactored in the future.

Reply:
This code path is currently covered by the HTTP/0.9 tests, but adding an explicit test sounds like a good idea to me!